libr/core/cmd_anal.c

```diff
@@ -4691,64 +4692,69 @@ static void cmd_anal_blocks(RCore *core, const char *input) {
 
 static void _anal_calls(RCore *core, ut64 addr, ut64 addr_end) {
 	RAnalOp op;
-	int bufi;
 	int depth = r_config_get_i (core->config, "anal.depth");
 	const int addrbytes = core->io->addrbytes;
 	const int bsz = 4096;
 	ut8 *buf;
-	ut8 *block;
-	bufi = 0;
+	ut8 *block0;
+	ut8 *block1;
+	int bufi = 0;
+	int bufi_max = bsz - 16;
 	if (addr_end - addr > UT32_MAX) {
 		return;
 	}
 	buf = malloc (bsz);
-	block = malloc (bsz);
-	if (!buf || !block) {
+	block0 = malloc (bsz);
+	block1 = malloc (bsz);
+	if (!buf || !block0 || !block1) {
 		eprintf ("Error: cannot allocate buf or block\n");
 		free (buf);
-		free (block);
+		free (block0);
+		free (block1);
 		return;
 	}
-
+	memset (block1, -1, bsz);
+	memset (block0, 0, bsz);
 	int minop = r_anal_archinfo (core->anal, R_ANAL_ARCHINFO_MIN_OP_SIZE);
 	if (minop < 1) {
 		minop = 1;
 	}
+	int setBits = r_config_get_i (core->config, "asm.bits");
+	r_cons_break_push (NULL, NULL);
 	while (addr < addr_end) {
 		if (r_cons_is_breaked ()) {
 			break;
 		}
 		// TODO: too many ioreads here
-		if (bufi > 4000) {
+		if (bufi > bufi_max) {
 			bufi = 0;
 		}
 		if (!bufi) {
-			r_io_read_at (core->io, addr, buf, bsz);
+			memset (buf, 0, bsz);
```

Comment:
why is this necessary? it seems to me `r_io_read_at` already does the `memset` with the io->ff value when necessary.

Reply:
removed the memset